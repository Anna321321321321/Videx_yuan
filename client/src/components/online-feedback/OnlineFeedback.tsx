import React, { Fragment } from 'react';
import { compose, withState } from 'recompose';
import * as Logger from 'videx/client/logger';
import FormDecorator from './FormDecorator';
import { Icon } from 'semantic-ui-react';

const enhance = compose(withState('visible', 'changeVisible', false));

interface OnlineFeedbackProps {
  visible: boolean;
  changeVisible: (visible: boolean) => void;
}

export const OnlineFeedback = (props: OnlineFeedbackProps) => (
  <Fragment>
    <Icon color="yellow" name="smile" />
    <a
      onClick={() => {
        Logger.event('OnlineFeedback.Open');
        props.changeVisible(true);
      }}
    >
      Online Feedback
    </a>
    <FormDecorator
      visible={props.visible}
      onCancel={() => props.changeVisible(false)}
    />
  </Fragment>
);

export default enhance(OnlineFeedback);
