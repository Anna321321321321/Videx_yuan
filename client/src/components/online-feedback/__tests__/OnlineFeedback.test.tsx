import { shallow } from 'enzyme';
import React from 'react';
import { OnlineFeedback } from '../OnlineFeedback';

describe('OnlineFeedback Unit Tests', () => {
  it('hidden modal', () => {
    const wrapper = shallow(
      <OnlineFeedback visible={false} changeVisible={null} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('display modal', () => {
    const wrapper = shallow(
      <OnlineFeedback visible={true} changeVisible={null} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
