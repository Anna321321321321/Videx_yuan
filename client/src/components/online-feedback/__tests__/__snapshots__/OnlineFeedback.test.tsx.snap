// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlineFeedback Unit Tests display modal 1`] = `
<React.Fragment>
  <Icon
    as="i"
    color="yellow"
    name="smile"
  />
  <a
    onClick={[Function]}
  >
    Online Feedback
  </a>
  <withState(withState(withHandlers(Component)))
    onCancel={[Function]}
    visible={true}
  />
</React.Fragment>
`;

exports[`OnlineFeedback Unit Tests hidden modal 1`] = `
<React.Fragment>
  <Icon
    as="i"
    color="yellow"
    name="smile"
  />
  <a
    onClick={[Function]}
  >
    Online Feedback
  </a>
  <withState(withState(withHandlers(Component)))
    onCancel={[Function]}
    visible={false}
  />
</React.Fragment>
`;
